src/virtual-source-buffer.js

```diff
@@ -247,29 +240,33 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
     let types = ['audio', 'video'];
 
     types.forEach((type) => {
+      // Don't create a SourceBuffer of this type if we don't have a
+      // codec for it
       if (!this[`${type}Codec_`]) {
         return;
       }
 
-      // if the mediasource already has a buffer for the codec
-      // use that
-      if (this.mediaSource_[`${type}Buffer_`]) {
-        this[`${type}Buffer_`] = this.mediaSource_[`${type}Buffer_`];
+      // Do nothing if a SourceBuffer of this type already exists
+      if (this[`${type}Buffer_`]) {
```

Comment:
I feel that the template strings, while clever and a nice way to reduce the code, make this code hard enough to reason about that I wanted to keep the conditions under which we exit early explicit. Also there is no good way to have two comments if we combine the conditions into one expression.